stream_buffer.c

```diff
@@ -437,25 +526,31 @@ BaseType_t xStreamBufferReset( StreamBufferHandle_t xStreamBuffer )
     /* Can only reset a message buffer if there are no tasks blocked on it. */
     taskENTER_CRITICAL();
     {
-        if( pxStreamBuffer->xTaskWaitingToReceive == NULL )
+        if( ( pxStreamBuffer->xTaskWaitingToReceive == NULL ) && ( pxStreamBuffer->xTaskWaitingToSend == NULL ) )
         {
-            if( pxStreamBuffer->xTaskWaitingToSend == NULL )
-            {
-                prvInitialiseNewStreamBuffer( pxStreamBuffer,
-                                              pxStreamBuffer->pucBuffer,
-                                              pxStreamBuffer->xLength,
-                                              pxStreamBuffer->xTriggerLevelBytes,
-                                              pxStreamBuffer->ucFlags );
-                xReturn = pdPASS;
-
-                #if ( configUSE_TRACE_FACILITY == 1 )
-                    {
-                        pxStreamBuffer->uxStreamBufferNumber = uxStreamBufferNumber;
-                    }
-                #endif
-
-                traceSTREAM_BUFFER_RESET( xStreamBuffer );
-            }
+            #if ( configUSE_SB_COMPLETED_CALLBACK == 1 )
+                {
+                    pxSendCallback = pxStreamBuffer->pxSendCompletedCallback;
+                    pxReceiveCallback = pxStreamBuffer->pxReceiveCompletedCallback;
+                }
+            #endif
+
+            prvInitialiseNewStreamBuffer( pxStreamBuffer,
```

Comment:
I do not understand this - why do we need these local variables? Can it not just be:

```
prvInitialiseNewStreamBuffer( pxStreamBuffer,
                              pxStreamBuffer->pucBuffer,
                              pxStreamBuffer->xLength,
                              pxStreamBuffer->xTriggerLevelBytes,
                              pxStreamBuffer->ucFlags,
                              pxStreamBuffer->pxSendCompletedCallback,
                              pxStreamBuffer->pxReceiveCompletedCallback );
```

Reply:
Its required since `pxStreamBuffer->pxSendCompletedCallback` and `pxStreamBuffer->pxReceiveCompletedCallback`  is not part of struct when `configUSE_SB_COMPLETED_CALLBACK` is `0`